apps/backend/migrations/20230725092535-change-group-names-to-unique.js

```diff
@@ -0,0 +1,77 @@
+'use strict';
+
+module.exports = {
+  up: async (queryInterface, Sequelize) => {
+    await queryInterface.sequelize.query(`
+    CREATE OR REPLACE PROCEDURE update_group_names_transaction()
+    LANGUAGE plpgsql
+    AS $$
+    DECLARE
+      check_name boolean := true;
+      offSet_cnt INTEGER :=1;
+      processing_group TEXT := '';
+      rec_group RECORD;
+      dup_names NO SCROLL CURSOR FOR
+        SELECT DISTINCT id, name FROM "Groups" WHERE name IN
+          (SELECT name FROM(SELECT id, name, ROW_NUMBER() OVER(PARTITION BY name) AS occurrences FROM "Groups") AS duplicates
+        WHERE duplicates.occurrences > 1) ORDER BY name, id;
+
+    BEGIN
+      -- Open the cursor
+      OPEN dup_names;
+      LOOP
+        -- retrieve a row of data from the result set (cursor) into the rec_group variable
+        FETCH dup_names INTO rec_group;
+      -- exit when no more records are available in the resul set
+        EXIT WHEN NOT FOUND;
+
+      -- Skip the first duplicate name (we only want to update subsequent dup names)
+        IF rec_group.name = processing_group THEN
+        -- Determine the next non-duplicate name
+          WHILE check_name = true LOOP
+            IF (SELECT EXISTS (SELECT name FROM "Groups" WHERE name = CONCAT(rec_group.name, '-', offSet_cnt))::int = 1) THEN
+              offSet_cnt := offSet_cnt + 1;
+            ELSE
+              check_name := false;
+            END IF;
+          END LOOP;
+          -- Uncomment the next line when debugging
+          -- RAISE NOTICE 'Updating % id % to %', rec_group.name, rec_group.id, CONCAT(rec_group.name, '-', offSet_cnt);
+
+        -- Update the table (comment when debugging)
+          UPDATE "Groups" SET name = CONCAT(rec_group.name, '-', offSet_cnt) WHERE id = rec_group.id;
+
+        -- reset the controlling variables
+          offSet_cnt :=1;
+          check_name = true;
+      -- Set the group name flag to the name we are processing
+        ELSE
+          -- Uncomment the next line when debugging
+          -- RAISE NOTICE 'Processing % id %', rec_group.name, rec_group.id;
+
+        -- Update the controlling processing name variable
+          processing_group := rec_group.name;
+        END IF;
+      END LOOP;
+      -- close the cursor
+      CLOSE dup_names;
+    END; $$
+    `)
+    await queryInterface.sequelize.query(`CALL update_group_names_transaction()`)
```

Comment:
why are we writing the transaction and then calling it as opposed to using sequelize.transaction?

Reply:
why would it be necessary to fix them again?  this is just to transition the group names the one time.  after that, our systems should be able to enforce the uniqueness requirement.  if the user goes in and manually mucks around with the group names within the db itself, that's on them to fix.